test/test_torch.py

```diff
@@ -2901,18 +2901,58 @@ def test_tensor_factory_cuda_type(self):
         self.assertTrue(x.is_cuda)
         torch.set_default_tensor_type(saved_type)
 
+    def test_tensor_randoms_bool(self):
```

Comment:
I feel a little uncomfortable about the fact that we are not actually testing against the most likely bug, which is that we are generating a number uniformly at random from 0-255 and then casting it to bool (which means that true would be 255x more likely to show up than false). We're not testing this because you never actually look at the values in question.

I'd suggest determining some acceptable level of flakiness, and then explicitly testing the distribution up to some p-value. As we seed before all tests, the test will deterministically pass or fail (unless someone perturbs the test itself; thus the acceptable level of flakiness.) An example is in this test:

```
    def test_random(self):
        # This test is flaky with p<=(2/(ub-lb))^200=6e-36
        t = torch.FloatTensor(200)
        lb = 1
        ub = 4

        t.fill_(-1)
        t.random_(lb, ub)
        self.assertEqual(t.min(), lb)
        self.assertEqual(t.max(), ub - 1)

        t.fill_(-1)
        t.random_(ub)
        self.assertEqual(t.min(), 0)
        self.assertEqual(t.max(), ub - 1)
```

Reply:
Ok, removing bool support for all randoms except the randint.